contributors/guide/scalability-good-practices.md

```diff
@@ -102,68 +96,62 @@ func (c *ControllerX) Run() {
+The first time, this was implemented in a very simple way, rapidly making the scheduler unusable, and `Pod` startup times went through the roof. We were forced to block this feature, and it did not make into the target release. Later, we slightly improved the algorithm to `O(#(scheduled Pods with PodAntiAffinity) * #Nodes)`, which was enough to allow the feature to get in as beta, with a huge asterisk next to it.
  
-This should serve as an example for the fact that we're working in a very complex area, where a lot of problems are actually very complex - not only non-linear, but some of them are NP-complete. Because of that it's understandable that you need to write something complex. But when you do that you need to take care to protect the rest of the system from that complexity, and add it only in the place where it's absolutely necessary.
+This example illustrates how we are working in a very complex area, where many problems can be much more complex than they seem. Not only that, they are non-linear, and some of them are NP-complete. Understandably, sometimes you need to write something complex, but when you do, you must protect the rest of the system from that complexity, and add it only where it is absolutely necessary.
 
 ### Big dependency changes
-*TL;DR:* When updating big dependencies (especially etcd or golang version) you need to run large scale test to verify that bump doesn't break Kubernetes performance.
- 
-Kubernetes depends on pretty much the whole universe. From time to time we need to update some dependencies (Godeps, etcd, go version). This can break us in many ways, and we saw it doing it a couple of times now. We skipped one version of `go` (1.5) exactly because of breaking our performance, and in the moment of writing this doc we're working with go team on understanding why `go1.8` also makes Kubernetes performance noticeably worse.
+Kubernetes depends on pretty much the whole universe. From time to time we need to update some dependencies (Godeps, etcd, go version). This can break us in many ways, as has already happened a couple of times. We skipped one version of `go` (1.5) precisely because it broke our performance. As this is being written, we are working with the go team to try to understand why `go1.8` negatively affects Kubernetes performance.
  
-If you're changing some big and important dependency, there's no way of knowing what performance impact it'll have. The only thing we can do is to actually run test and check.
+If you are changing a large and important dependency, there is no way of knowing what performance impact it will have. We can only run test and check.
 Where to look to get data?
```

Comment:
Consider making this a subheader

Reply:
What about the paragraph above it? Seems to be about the same subject.

Reply:
Point taken on the content. What I'm really after is visual separation. In the raw markdown text, this question is on its own line. If you look at the "pretty" output, however, you'll see that is not the case; it's just hidden inside the paragraph, which I don't think is the intention. Adding an extra blank line in the markdown will create a paragraph.

Reply:
Ok, I broke it out on its own line as a subheader and also made small changes to make the rest of the content in the section easier to read and understand (just formatting). Let me know if this works for you.